tofhir-server/src/main/scala/io/tofhir/server/service/SimpleStructureDefinitionService.scala

```diff
@@ -74,7 +74,12 @@ class SimpleStructureDefinitionService(fhirConfig: BaseFhirConfig) {
       }
 
       // Start of the simplifier method
-      if (profileUrl.isDefined && accumulatingTypeUrls.contains(profileUrl.get)) {
+      if (profileUrl.isDefined &&
+        // Stop the recursion if the profile is already visited twice (to avoid infinite recursion)
+        (accumulatingTypeUrls.count(_ == profileUrl.get) >= 2 ||
```

Comment:
I would implement an elimination to this so that it does not count (twice) every time. I will update this part.